datafusion/common/src/dfschema.rs

```diff
@@ -806,6 +820,12 @@ impl From<&DFSchema> for Schema {
     }
 }
 
+impl AsRef<Schema> for DFSchema {
```

Comment:
This allows DFSchema to be treated like a &Schema, which is now possible after https://github.com/apache/datafusion/pull/9595

Reply:
:exploding_head:

Reply:
Very nice!

Reply:
yeah, https://github.com/apache/datafusion/pull/9595 was epic -- kudos to @haohuaijin  for making that happen